spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -127,6 +129,15 @@ class SparkWrite {
     this.dsSchema = dsSchema;
     this.extraSnapshotMetadata = writeConf.extraSnapshotMetadata();
     this.partitionedFanoutEnabled = writeConf.fanoutWriterEnabled();
+
+    Preconditions.checkArgument(
```

Comment:
Nit:  This is a bit hard to read, would this be better?
```
if (writeConf.outputSpecId == null) {
  this.outputSpecId = table.spec().specId();
} else {
  this.outputSpecId = writeConf.outputSpecId();
  Preconditions.checkArgument(table.specs().containsKey(outputSpecId), ...);
}
```

Reply:
I agree, this is much simpler. The ternary operator became too unwieldy in this case. Done.